app/src/main/java/com/sacramento/movies/ui/search_movie_by_name/SearchMovieByNameViewModel.kt

```diff
@@ -44,7 +57,52 @@ class SearchMovieByNameViewModel @Inject constructor(
 
         lastFetchedMovieResult = newFetchedMoviesResult
 
-        _movies = newFetchedMoviesResult
+        return newFetchedMoviesResult
+    }
+
+    private  fun fetchMoviesByNameFormDB(): Flow<PagingData<MovieModel>>{
+
+        state.get<String>("movieName")?.let { moviesPagingSourceDB.setupMovieName(it) }
+
+        val lastMovieResult = lastFetchedMovieResultDB
+        if(lastMovieResult != null) return lastMovieResult
+
+
+        val newFetchedMoviesResult = Pager(config = PagingConfig(
+            pageSize = 100,
+            enablePlaceholders = false,
+            initialLoadSize = 100
+        ),
+            pagingSourceFactory = { moviesPagingSourceDB }
+        ).flow.cachedIn(viewModelScope)
+
+        lastFetchedMovieResultDB = newFetchedMoviesResult
+
+        return newFetchedMoviesResult
+    }
+
+    fun refreshList() {
+        lastFetchedMovieResult = null
+        lastFetchedMovieResultDB = null
+    }
+
+
+
+    private fun isNetworkAvailable(): Boolean {
```

Comment:
You have the same method in MoviesViewModel
It is better to create some separate class, like a ConnectionManager or ConnectionHelper, and move there this method.
Also, Hilt can inject application object right into this new object, and you can get rid of AndroidViewModel here.

Reply:
Fixed